project_metadata.go

```diff
@@ -0,0 +1,16 @@
+package lifecycle
+
+const (
+	ProjectMetadataLabel    = "io.buildpacks.project"
+	ProjectMetadataFileName = "project-metadata.toml"
+)
+
+type ProjectMetadata struct {
+	Source ProjectSource `toml:"source" json:"source"`
+}
+
+type ProjectSource struct {
+	Type     string      `toml:"type" json:"type"`
+	Version  interface{} `toml:"version" json:"version"`
+	Metadata interface{} `toml:"metadata" json:"metadata"`
+}
```

Comment:
Think we might need some `omitemptys` here if this is going on and off disk?

Reply:
Instead of omitting metadata or version when they are not supplied I would prefer if the type of `Version` and `Metadata` changed to  `map[string]interface{}` and we ensured they were never nil. If, for example there was no metadata provided, I would prefer to see `{..., "metadata": {}}` in my label to `{..., "metadata": null}`.

This suggestion better matches the RFC in which all example values for `version` and `metadata` keys are maps.

Reply:
Thanks for the comments @zmackie and @ekcasey 
I've amended `Version` and `Metadata` to be of type `map[string]interface{}` as it matches the intent of the RFC.